kernel/src/path.rs

```diff
@@ -299,6 +302,50 @@ impl ParsedLogPath<Url> {
         }
         Ok(path)
     }
+
+    /// Extract the In-Commit Timestamp from the CommitInfo action in this commit log file.
+    /// This is a utility function that can be used by multiple parts of the codebase
+    /// (snapshot, CDF, time travel, etc.).
+    ///
+    /// Returns the inCommitTimestamp value, or an error if ICT is not found or cannot be read.
+    /// Callers should handle enablement version checks before calling this method.
+    #[internal_api]
+    pub(crate) fn read_in_commit_timestamp(&self, engine: &dyn Engine) -> DeltaResult<i64> {
+        // Only works on commit files
+        if !self.is_commit() {
+            return Err(Error::generic(format!(
+                "read_in_commit_timestamp can only be called on commit files, got: {:?}",
+                self.file_type
+            )));
+        }
+
+        let file_meta = FileMeta {
+            location: self.location.clone(),
+            last_modified: 0, // Not used for reading
+            size: 0,          // Not used for reading
+        };
```

Comment:
how about we move this `read_in_commit_timestamp` to an `impl ParsedLogPath<FileMeta>` and not have to worry about faking the modtime/size for a raw Url